TinyME-main/src/test/java/ir/ramtung/tinyme/domain/SecurityTest.java

```diff
@@ -0,0 +1,139 @@
+package ir.ramtung.tinyme.domain;
+
+import ir.ramtung.tinyme.config.MockedJMSTestConfig;
+import ir.ramtung.tinyme.domain.entity.*;
+import ir.ramtung.tinyme.messaging.exception.InvalidRequestException;
+import ir.ramtung.tinyme.messaging.request.DeleteOrderRq;
+import ir.ramtung.tinyme.messaging.request.EnterOrderRq;
+import ir.ramtung.tinyme.domain.service.Matcher;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Import;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+@SpringBootTest
+@Import(MockedJMSTestConfig.class)
+class SecurityTest {
+    private Security security;
+    private Broker broker;
+    private Shareholder shareholder;
+    private List<Order> orders;
+    @Autowired
+    Matcher matcher;
+    @BeforeEach
+    void setupOrderBook() {
+        security = Security.builder().build();
+        broker = Broker.builder().brokerId(0).credit(1_000_000L).build();
+        shareholder = Shareholder.builder().shareholderId(0).build();
+        shareholder.incPosition(security, 100_000);
+        orders = Arrays.asList(
+                new Order(1, security, Side.BUY, 304, 15700, broker, shareholder),
+                new Order(2, security, Side.BUY, 43, 15500, broker, shareholder),
+                new Order(3, security, Side.BUY, 445, 15450, broker, shareholder),
+                new Order(4, security, Side.BUY, 526, 15450, broker, shareholder),
+                new Order(5, security, Side.BUY, 1000, 15400, broker, shareholder),
+                new Order(6, security, Side.SELL, 350, 15800, broker, shareholder),
+                new Order(7, security, Side.SELL, 285, 15810, broker, shareholder),
+                new Order(8, security, Side.SELL, 800, 15810, broker, shareholder),
+                new Order(9, security, Side.SELL, 340, 15820, broker, shareholder),
+                new Order(10, security, Side.SELL, 65, 15820, broker, shareholder)
+        );
+        orders.forEach(order -> security.getOrderBook().enqueue(order));
+    }
```

Comment:
setup is repeated in all test classes. It is better to have an interface and define a default method in it